src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp

```diff
@@ -1424,14 +1443,35 @@ namespace OpenMS
     {
       double cummin = 1.0;
 
-      for (auto&& rit = scores_to_FDR.begin(); rit != scores_to_FDR.end(); ++rit)
+      if (higher_score_better)
+      {
+        for (auto&& rit = scores_to_FDR.begin(); rit != scores_to_FDR.end(); ++rit)
+        {
+        #ifdef FALSE_DISCOVERY_RATE_DEBUG
+          std::cerr << "Comparing " << rit->second << " to " << cummin << std::endl;
+        #endif
+          cummin = std::min(rit->second, cummin);
+          rit->second = cummin;
+        }
+      }
+      else
       {
+        //TODO this does not solve the ultimate problem of setting the scores, since setScores always uses lower_bound.
+        // Which for higher_score_better yields the correct "greater or equal = better or equal". But for
+        // lower_score_better this yields "greater or equal = worse or equal".
+        // Two options now: revert current changes and wrap map<K,V,correctcomparator> to something that one can pass to all
+        // the set methods and lower_bound gives always the right element.
+        // Pass OLD score (i.e. the score in score_to_fdr) higher_better sorting to the set methods
```

Comment:
would it be possible to implement the lower score better using a transformation in the beginning and backtransformation in the end?
e.g., max_score - score